external/src/main/java/com/redhat/cloud/policies/engine/actions/plugins/NotificationActionPluginListener.java

```diff
@@ -44,72 +51,103 @@
     public static final String APP_NAME = "policies";
     public static final String EVENT_TYPE_NAME = "policy-triggered";
 
+    private static final Logger log = Logger.getLogger(NotificationActionPluginListener.class.getName());
+    private final ConcurrentSkipListMap<String, PoliciesAction> notifyBuffer = new ConcurrentSkipListMap<>();
```

Comment:
I should have asked that earlier, but why do we need to take care of concurrency here? I would expect a `@Dependent` bean to be short-lived and used by a single thread. Can you tell me more about this class lifecycle and why it is used by multiple threads?

Reply:
That's a good question, I don't know the answer to it. 
I was repeating a pattern for doing the aggregation found in an older version of the code: https://github.com/RedHatInsights/policies-engine/blob/e4fb24c768e0309332536718e7bea3674fa79a11/external/src/main/java/com/redhat/cloud/policies/engine/actions/plugins/EmailActionPluginListener.java#L50

From my understanding, this class is instantiated at code initialization only once, and set to handle all the `policies` that matches the condition of a system payload (a system check-in to insights). So I don't think is short-lived and used by a single thread.

I still don't know much about the inner workings of [hawkular-alerts](https://github.com/hawkular/hawkular-alerts) (the project which this one forked from) or the engine itself. I'll try to seek better answers for this.

Reply:
It looks like the bean should be `@ApplicationScoped` instead of `@Dependent`. It's not related to this PR though so I'm resolving this discussion.